src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -33,11 +42,66 @@ def cli_asset():
     default=False,
 )
 @_handle_error
-def asset_snapshot(**args):
+def asset_snapshot(root_dir, manifest_out, recursive, **args):
     """
     Creates manifest of files specified root directory.
     """
-    click.echo("snapshot taken")
+    root_dir_basename = os.path.basename(root_dir) + "_"
+
+    if not os.path.isdir(root_dir):
+        misconfigured_directories_msg = f"Specified root directory {root_dir} does not exist. "
+        raise NonValidInputError(misconfigured_directories_msg)
+
+    if manifest_out and not os.path.isdir(manifest_out):
+        misconfigured_directories_msg = (
+            f"Specified destination directory {manifest_out} does not exist. "
+        )
+        raise NonValidInputError(misconfigured_directories_msg)
+    elif manifest_out is None:
+        manifest_out = root_dir
+
+    inputs = []
+    for root, dirs, files in os.walk(root_dir):
+        if os.path.basename(root).endswith("_manifests"):
+            continue
```

Comment:
yes, the assumption is that users wont name things ending in `_manifest`, I will make this more robust and check for `root-dir-name` + `manifest`, but I think it's important to stay because subdirectories can also have manifests, and I think its uneccessary to upload it as file data, if it's already being uploaded as a manifest

also `--manifest-out` shouldn't change things, because users could do a first snapshot on the rootdir without specifying, then a second time with `--manifest-out`, and the manifests folder should still be ignored

Reply:
spoke with jericht, will remove for now, as enforcing files named manifest may be unsafe